Vizzuality/heco-invest
# Adding regions to Colombia locations

Second part of feature focused on import of location data. This one is about Colombia regions (mosaics) ;)

I have added new model `LocationMember` which allows to have `have_many` relation between other locations (regions). Reason for this is that administrative location (state, municipality, etc.) can be at multiple geographic locations (regions). Locations importer now supports import of regions and also assigns these regions to correct municipalities.

About regions data -- I have used shapefile provided via Jira ticket to get regions and downloaded coordinates for every municipality from wikipedia. Then I have computed intersections and decided which municipalities are at which regions. Unfortunately, this approach is super inaccurate, because whole municipality is represented just by one point and even though point itself is not at region, part of the municipality still can be. Unfortunately, I don't have exact geometry for every municipality so this is as far as I can go with wikipedia data.
 
![colombia_regions](https://user-images.githubusercontent.com/20660167/159039565-ff1d7756-4ad8-4b16-a56a-8634792cbce9.png)

## Testing instructions

Same as first iteration of location import feature. Run `rake db:seed` and check that LocationMember table contains data ;)

## Tracking

https://vizzuality.atlassian.net/browse/LET-130
